### ISSUE TITLE
FPS tanks when using shaders on mesa 20.1.3-1

### ISSUE THREAD

**Issue by KermitTheFrogerino** [sp614x/optifine#4427]
The performance tanks down to around 20 fps on my paper server (not on this PC). I've got a Radeon 5700XT, Ryzen 7 1700 3.9Ghz overclock, 3000mhz 16 of ram, Manjaro with kernel 5.7.9-1-MANJARO running on X11. I've got the same issue in Windows too. The GPU usage is around 90% while the CPU is at 10% (one core pegged at 80-90%, only without shaders). With Shaders, the CPU usage is around 30% which could indicate a CPU bottleneck. Never really had this issue be this extreme on the 1.15 release but it was still there. I usually get 25fps when running with 12 chunks and 60 when at 8 chunks.

I've tried the latest BSL shaders, Sildurs enhanced default (lowest settings), SEUS Renewed (which doesn't work but that's a different issue...), Sildurs Vibrant Shaders Extreme VL. All of them experience performance issues. 

I'm running at 12 chunks and sometimes 8 chunks (this could be the reason but idk. the difference between these is 84585600 blocks which is insane). My Optifine version is "OptiFine 1.16.1 HD U G2 pre9", installed through the jar file directly as a new profile (no forge, no mods).

![image](https://user-images.githubusercontent.com/35975961/88324291-c8861000-cd23-11ea-9baf-439ebf8fa81b.png)
Running on the BSL shader.

![image](https://user-images.githubusercontent.com/35975961/88324834-8e693e00-cd24-11ea-96d5-843bb2d016e0.png)
The stats from MangoHUD

**Comment by unicornbloods** [comment]
Is this still an issue on the full release of optifine 1.16.1?

**Comment by KermitTheFrogerino** [comment]
@unicornbloods Yeah but only in Linux though. Render regions doesn't work with the MESA drivers. With it enabled, the FPS tanks while in Windows it works as expected

**Comment by unicornbloods** [comment]
> @unicornbloods Yeah but only in Linux though. Render regions doesn't work with the MESA drivers. With it enabled, the FPS tanks while in Windows it works as expected

Okie thank you for the reply!!!! Can you close this thread now?

**Comment by KermitTheFrogerino** [comment]
@unicornbloods why? The issue isn't solved

**Comment by nekkan** [comment]
I am having the same problem on a Ryzen 3 3200G + RX 550, Ubuntu 20.10.

**Comment by ErikReider** [comment]
> I am having the same issue on a Ryzen 3 3200G + RX 550, Ubuntu 20.10.

I fixed the issue a few months ago by running a script that enables some Mesa variables. I've posted the script here

https://github.com/ErikReider/Minecraft-Optifine-Shaders-Linux-fix

**Comment by nekkan** [comment]
> > I am having the same issue on a Ryzen 3 3200G + RX 550, Ubuntu 20.10.
> 
> I fixed the issue a few months ago by running a script that enables some Mesa variables. I've posted the script here
> 
> https://github.com/ErikReider/Minecraft-Optifine-Shaders-Linux-fix

Thank you so much! Alternatively, in MultiMC you can just set the wrapper command to `bash -c "export vblank_mode=0;export force_glsl_extensions_warn=true;export mesa_glthread=true;gamemoderun $INST_JAVA \"$@\""` (or `bash -c "export vblank_mode=0;export force_glsl_extensions_warn=true;export mesa_glthread=true;$INST_JAVA \"$@\""` if you aren't using the Feral Gamemode).
